Add Free Bus command

nosjojo opened the pull request:
Adds a call to access the Free Bus function of the Aardvark.

I opted to check the status code on return, which will cause an exception to be raised if the bus was already free. Even though it's a non-issue (bus is free on either outcome), the user won't be able to determine that the bus _was_ already free, which could potentially be valuable to the user. For library code, I prefer not to remove choices from users.

Realistically, all usages of the function will likely be:
```python
try:
    dev.i2c_free_bus()
except IOError as e:
    if (e.errno == ERR_I2C_BUS_ALREADY_FREE):
        pass
    else:
        raise e
```

mwalle commented on pyaardvark/aardvark.py:
```diff
@@ -447,6 +447,15 @@ def i2c_master_write_read(self, i2c_address, data, length):
         self.i2c_master_write(i2c_address, data, I2C_NO_STOP)
         return self.i2c_master_read(i2c_address, length)
 
+    def i2c_free_bus(self):
```
please name it i2c_stop()

also add a parameter ignore_errors=True; if set, ignore the ERR_I2C_BUS_ALREADY_FREE error.

That way it can be used like
try:
  dev.i2c_master_write_read()
except IOError:
  dev.i2c_stop()

mwalle reviewed (commented):
Thanks for your patch!

Here are some things to improve:

- please add a proper commit message
- add a Signed-off-by: line
- add a test case for the new function, see tests/
- finally there are bonus points if you'd add the try except handling to the (or a new?) example in docs/intro.rst

nosjojo commented:
> Thanks for your patch!
> 
> Here are some things to improve:
> 
> * please add a proper commit message
> * add a Signed-off-by: line
> * add a test case for the new function, see tests/
> * finally there are bonus points if you'd add the try except handling to the (or a new?) example in docs/intro.rst

- For the commit messages, I didn't think about it until after I modified the branch, so I can't amend the original commit, it requires a force push to do that. However, when you merge the branch, you can select Squash and Merge from the menu. That will let you modify the final message to include any adjustments. The 2nd commit was properly detailed, so it just needs an extra line regarding the addition of the `i2c_stop` command.
- The signed-off-by functionality is integrated into github (doing these edits on the website). My individual commits are showing as _verified_ which is equivalent to the `git commit -S` usage. I can add a string to the PR message, but doesn't seem necessary. 
- For the test cases: I'm not familiar with nose tests, so I'm not 100% sure those are written correctly, but I added 3.
- I added a reference to the new command to the `intro.rst` file, but I didn't go into detail regarding the exception handling. Instead, I just left the reference with the `ignore_errors` parameter set to True. I also fixed 2 typos I spotted.

mwalle commented:
Hi looks good so far except the SoB tag, please note that "git commit -S" is not what is adding the SoB, but it is "git commit -s". For the two commits, yes I can certainly squash that into one. So please add the SoB with your real name and email and please also add yourself to AUTHORS.

Thanks,
-michael

nosjojo commented:
> Hi looks good so far except the SoB tag, please note that "git commit -S" is not what is adding the SoB, but it is "git commit -s". For the two commits, yes I can certainly squash that into one. So please add the SoB with your real name and email and please also add yourself to AUTHORS.
> 
> Thanks, -michael

Should be good to go now.

mwalle commented:
Please have a look at the master branch and tell me if it is ok as it is. Then I'll close this MR.

nosjojo commented:
> Please have a look at the master branch and tell me if it is ok as it is. Then I'll close this MR.

No issues spotted, should be good to go.
